android/src/main/java/com/bitmovin/player/reactnative/BufferModule.kt

```diff
@@ -32,36 +32,24 @@ class BufferModule(private val context: ReactApplicationContext) : ReactContextB
                 player.buffer.getLevel(bufferType, MediaType.Audio),
                 player.buffer.getLevel(bufferType, MediaType.Video),
             )
-            JsonConverter.fromRNBufferLevels(bufferLevels).let {
-                promise.resolve(it)
-            }
+            promise.resolve(bufferLevels.toJson())
         }
     }
 
     /**
      * Sets the target buffer level for the chosen buffer type across all media types.
      * @param nativeId Target player id.
-     * @param type The [type of buffer][JsonConverter.toBufferType] to set the target level for.
+     * @param type The [type of buffer][toBufferType] to set the target level for.
      * @param value The value to set.
      */
     @ReactMethod
-    fun setTargetLevel(nativeId: NativeId, type: String, value: Double) {
-        uiManager()?.addUIBlock { _ ->
-            val player = playerModule()?.getPlayer(nativeId) ?: return@addUIBlock
-            val bufferType = JsonConverter.toBufferType(type) ?: return@addUIBlock
+    fun setTargetLevel(nativeId: NativeId, type: String, value: Double, promise: Promise) {
+        addUIBlock(promise) {
+            val player = playerModule().getPlayer(nativeId) ?: return@addUIBlock
+            val bufferType = type.toBufferType() ?: return@addUIBlock
```

Comment:
Can we simplify the return?
```suggestion
            val bufferType = type.toBufferType() ?: return
```